sdk/core/azure_core_amqp/src/fe2o3/connection.rs

```diff
@@ -53,64 +53,71 @@ impl AmqpConnectionApis for Fe2o3AmqpConnection {
         options: Option<AmqpConnectionOptions>,
     ) -> Result<()> {
         {
+            let options = options.unwrap_or_default();
+            let mut endpoint = url.clone();
```

Comment:
Does this actually need to be cloned? It's already owned at this point, and looking below it doesn't seem it needs to be either. Or is it just because you need it `mut`?

Reply:
Unfortunately, It's used on line 116 (`builder = builder.hostname(url.host_str))`).